server/Common/Api/QueryParameters.cs

```diff
@@ -5,17 +5,43 @@ namespace Tyger.Common.Api;
 
 public static class QueryParameters
 {
-    public static Dictionary<string, string>? GetTagQueryParameters(this HttpContext context)
+    public static Dictionary<string, string>? GetTagQueryParameters(this HttpContext context, string prefix = "tag")
     {
         Dictionary<string, string>? tagQuery = null;
         foreach (var tag in context.Request.Query)
         {
-            if (tag.Key.StartsWith("tag[", StringComparison.Ordinal) && tag.Key.EndsWith(']') && tag.Value.Count > 0)
+            if (tag.Key.StartsWith(prefix + "[", StringComparison.Ordinal) && tag.Key.EndsWith(']') && tag.Value.Count > 0)
             {
-                (tagQuery ??= []).Add(tag.Key[4..^1], tag.Value.FirstOrDefault() ?? "");
+                var start = prefix.Length + 1;
+                (tagQuery ??= []).Add(tag.Key[start..^1], tag.Value.FirstOrDefault() ?? "");
             }
         }
 
         return tagQuery;
     }
+
+    /// <summary>
+    /// Parses the "ttl" query parameter from the request and validates it.
+    /// </summary>
+    /// <returns>true if TTL was parsed or not provided, false if the TTL is invalid</returns>
+    public static bool ParseAndValidateTtlQueryParameter(this HttpContext context, out TimeSpan? ttl)
+    {
+        ttl = null;
+        if (context.Request.Query.TryGetValue("ttl", out var ttlValues))
+        {
+            if (!TimeSpan.TryParse(ttlValues, out var ttlParsed))
```

Comment:
Should this a bad request if the value is invalid?

Reply:
Yes it returns false if the `ttl` query parameter is invalid, and in `Buffers.cs` that results in a `400 Bad Request`.